### DISCUSSION TITLE
[feature request] Locking and dumping by database instead of global

### DISCUSSION THREAD

**Discussion by aqueos** [mydumper/mydumper#1378] [Ideas]
Hi,

  I have quite some issue on locking in mydumper blocking websites for too long. One big DB will block all the other's website.

  As you know a lot of project use a simple 1:1 way for database usage. I mean :

one application = 1 database, 
1 website =1 database. 

  Therefor to minimize lock time it would be very useful to have an option so that mydumper does a  lock by database and not a global one. 

  Meaning mydumper will process and lock the data database by database, no global lock, one only on the DB we work on. We can still parallel the dump but do it database by databse instead of random tables scatered on all DB with a global read only lock. 

 Otherwise i would need to do launch a bash script in a loop, to do a mydumper separated for each database but this is quite a lot of work to do and restore would need also a special loop for restoring the whole databases (not speaking of stupid DB name that can screw the bash script). Would be quite handy to have this database by database locking mecanism in the mydumper code.

  What do you think of this option, is that even possible in the code ?

best regards,
Ghislain.

**Comment by davidducos** [comment]
Hi @aqueos, if you want to avoid blocking the database AND you are using innodb, you can use --trx-consistency-only to reduce the time that FTWRL will be running. Or if you are using Percona Server and GTID on, you might be able to take safely a consistent backup with --no-locks.

**Comment by aqueos** [comment]
hi David,

I have quite a mix of case so some use legacy myisam or not. So for a lot of case i cannot use innodb only solutions :(

I dont use GTID because i have no replication setup but i use percona server. I dont seems to find any documentations about GTID implication for a single node setup (all the docs speak about replication usage of it ) and you say 'might' even if GTID is ON so i don't feel very sure about this if i find a innodb only server :p 

best regards,
Ghislain.
